source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -4200,3 +4200,13 @@ void Graph::writeText(std::string fileName, mx::FilePath filePath)
     writeOptions.elementPredicate = getElementPredicate();
     mx::writeToXmlFile(_graphDoc, filePath, &writeOptions);
 }
+
+template <typename Element>
+void Graph::setName(const Element& node, const std::string &name) {
```

Comment:
I'd recommend making this first argument an `ElementPtr`, making it more consistent with the broader codebase, and removing the need for a templated function.  Usually we pass `ElementPtr` objects by value rather than by reference in MaterialX, since these objects are relatively lightweight, and we've found compilers to be very capable in minimizing the overhead of passing shared pointers between functions.

Reply:
Hey so I actually have this as a templated function because it gets used by two different scenarios. 

There exists a different `setName` method inside of `UiNode` also that gets used. Since those aren't `mx::ElementPtr`s, i decided to use templates here and have all calls just go through this method. 

It would seem that both are relevant to the bug at hand here. I can modify `UiNode`s setName as well but I decided against it to avoid having to include `Element.h` (where `createValidChildName` comes from). 

What do you think?

Reply:
Hi @ZelboK,
I think the UI node setName does not need to be covered here as it should come from the Element itself, so sanitizing that should be enough. 

The other note is that I though in the issue that the easiest thing to do would be to change the actual Element::setName() method ?

Reply:
I was under the impression that rather than changing the core API, sanitizing the input was the way to go

Reply:
@ZelboK @kwokcb I'm open to either approach, though I do think that a fix to `Element::setName` will eventually be needed, so that other applications (both open and proprietary) are protected against similar issues cropping up.

If you'd like to tackle the fix directly in `setName`, that would be fine with me, or you can proceed along the current lines with a specific fix for the graph editor.

Reply:
@ZelboK Taking the broader view, I think that the fix to `Element::setName` would be the simplest, most robust change here, so I'd recommend taking that approach if that works for you.

I believe the gist of the change in `Element::setName` would be to check for an empty input name, and to call `createValidChildName` from the parent of the given element, in a similar manner to this logic:
https://github.com/AcademySoftwareFoundation/MaterialX/blob/c34a1066bb194d36174454b3b763a6d68bc8480d/source/MaterialXCore/Element.cpp#L229

If no parent element is found in `Element::setName`, I believe it's fine to allow the empty name string to be used without modification, as it's common for the root element of a document to be unnamed.